storm-core/src/jvm/backtype/storm/scheduler/resource/ResourceAwareScheduler.java

```
@@ -28,94 +32,270 @@
+            double assignedMemOffHeap = 0.0;
+            double assignedCpu = 0.0;
+
+            Set<String> nodesUsed = new HashSet<String>();
+            for (Map.Entry<WorkerSlot, Collection<ExecutorDetails>> workerToTasksEntry : schedulerAssignmentMap.entrySet()) {
+                WorkerSlot targetSlot = workerToTasksEntry.getKey();
+                Collection<ExecutorDetails> execsNeedScheduling = workerToTasksEntry.getValue();
+                RAS_Node targetNode = this.nodes.getNodeById(targetSlot.getNodeId());
+                targetNode.assign(targetSlot, td, execsNeedScheduling);
+                LOG.debug("ASSIGNMENT    TOPOLOGY: {}  TASKS: {} To Node: {} on Slot: {}",
+                        td.getName(), execsNeedScheduling, targetNode.getHostname(), targetSlot.getPort());
+                if (!nodesUsed.contains(targetNode.getId())) {
+                    nodesUsed.add(targetNode.getId());
+                }
+                assignedMemOnHeap += targetSlot.getAllocatedMemOnHeap();
+                assignedMemOffHeap += targetSlot.getAllocatedMemOffHeap();
+                assignedCpu += targetSlot.getAllocatedCpu();
+            }
+
+            Double[] resources = {requestedMemOnHeap, requestedMemOffHeap, requestedCpu,
+                    assignedMemOnHeap, assignedMemOffHeap, assignedCpu};
+            LOG.debug("setResources for {}: requested on-heap mem, off-heap mem, cpu: {} {} {} " +
+                            "assigned on-heap mem, off-heap mem, cpu: {} {} {}",
+                    td.getId(), requestedMemOnHeap, requestedMemOffHeap, requestedCpu,
+                    assignedMemOnHeap, assignedMemOffHeap, assignedCpu);
+            this.cluster.setResources(td.getId(), resources);
+            updateSupervisorsResources(this.cluster, this.topologies);
+            return true;
+        } else {
+            LOG.warn("schedulerAssignmentMap for topo {} is null. This shouldn't happen!", td.getName());
+            return false;
```

This seems like an error.  Should we throw an exception here?